kwave/kspaceFirstOrder2D.py

```diff
@@ -438,10 +434,8 @@ def kspaceFirstOrder2D(
         if options.save_to_disk_exit:
             return
 
-        input_filename = k_sim.options.save_to_disk
-        output_filename = os.path.join(tempfile.gettempdir(), 'output.h5')
-
-        executor = Executor(device='gpu')
+        executor = Executor(simulation_options=simulation_options, execution_options=execution_options)
         executor_options = execution_options.get_options_string(sensor=k_sim.sensor)
-        sensor_data = executor.run_simulation(input_filename, output_filename, options=executor_options)
-        return k_sim.sensor.combine_sensor_data(sensor_data)
```

Comment:
Noticed that you do not cal `combine_sensor_data(...)` anymore, neither here nor in the `run_simulation(...)` method. Don't we need it?

Reply:
sensor_data is now a dictionary of all variables output from the kspaceFirstOrder executable. Combine sensor data can only be applied to some values in that dictionary. This is the behavior of k-wave in MATLAB as well. In the example, an extra call is made to `combine_sensor_data(...)` on the `'p'` variable [here](https://github.com/waltsims/k-wave-python/blob/66ff3fcb15467a67b2cefacd2bb2fb17a0cae265/examples/bmode_reconstruction_example.py#L131C1-L132C100). Though this might not be optimal, it allows for greater functionality at the moment and is the behavior of MATLAB.